aws-api/src/main/java/com/amplifyframework/api/aws/AWSApiPlugin.java

```diff
@@ -710,14 +710,19 @@ private RestOperation createRestOperation(
             case GET:
             case DELETE:
                 if (options.hasData()) {
-                    throw new ApiException("HTTP method does not support data object! " + type,
-                            "Try sending the request without any data in the options.");
+                    operationRequest = new RestOperationRequest(
+                            type,
+                            options.getPath(),
+                            options.getData(),
+                            options.getHeaders(),
+                            options.getQueryParameters());
+                } else {
+                    operationRequest = new RestOperationRequest(
+                            type,
+                            options.getPath(),
+                            options.getHeaders(),
+                            options.getQueryParameters());
                 }
```

Comment:
With these changes, a GET or HEAD request would now be allowed to have a data object, which shouldn't be allowed. The original code under the DELETE case (that didn't allow a data object) can be added under `case GET:` to prevent a data object being added for GET or HEAD.